### PR TITLE
sys/newlib: Improve robustness in include path search

### PR DESCRIPTION
**Pull request by gebart**
This improves the robustness of the include path search when using non-standard newlib installation paths.
The fallback looks for the proper cross-gcc in the path of the current user and tries to find includes relative to that directory.

The include order patch is only relevant when building with Clang where we have to search for all the cross toolchain headers.

### CODE REVIEW

**Comment by gebart** [comment]
(Travis was idle)

File: sys/newlib/Makefile.include [hunk 1]
```diff
@@ -30,4 +30,11 @@ NEWLIB_INCLUDES ?= \
         $(foreach pat, $(NEWLIB_INCLUDE_PATTERNS), $(wildcard $(pat))), \
         -isystem $(dir))
 
-export INCLUDES += $(NEWLIB_INCLUDES)
+# If nothing was found we will try to fall back to searching for a cross-gcc in
+# the current PATH and use a relative path for the includes
+ifeq (,$(NEWLIB_INCLUDES))
+  NEWLIB_INCLUDES := $(addprefix -isystem ,$(wildcard $(dir $(shell which $(PREFIX)gcc))../$(TARGET_TRIPLE)/include))
```
Comment (authmillenon) [inline sys/newlib/Makefile.include]: What does `TARGET_TRIPLE` contain?

Comment (gebart) [inline sys/newlib/Makefile.include]: arm-none-eabi for cortex m, msp430-elf for msp430, etc.

**Comment by LudwigKnuepfer** [comment]
ACK

**Pull request merged**